src/h5web/dataset-visualizer/DimensionMapper.tsx

```diff
@@ -1,86 +1,78 @@
+                label: `D${dimIndex}`,
+                isSelected: mapperState[dimIndex] === 'x',
                 onClick: () => {
-                  if (mapping.x !== dimIndex) {
-                    const newMapping: DimensionMapping = {
-                      x: dimIndex,
-                      slicingIndices: { [mapping.x]: 0 },
-                    };
-                    onChange(newMapping);
+                  const prevX = mapperState.indexOf('x');
+                  if (prevX !== dimIndex) {
+                    const newMapperState = mapperState.slice();
+                    newMapperState[prevX] = 0;
+                    newMapperState[dimIndex] = 'x';
+                    onChange(newMapperState);
                   }
                 },
               };
             })}
           />
         </div>
         <div className={styles.sliderWrapper}>
-          {Object.entries(mapping.slicingIndices).map(
-            ([dimNumber, slicingIndex]) => (
-              <>
-                <button
-                  type="button"
-                  className={styles.btn}
-                  aria-pressed="true"
-                >
-                  {dimNumber}
-                </button>
+          {mapperState.map((slicingIndex, dim) =>
+            isNumber(slicingIndex) ? (
+              // eslint-disable-next-line react/no-array-index-key
+              <Fragment key={dim}>
+                <span className={styles.sliderLabel}>D{dim}</span>
                 <ReactSlider
-                  key={dimNumber}
                   className={styles.slider}
                   trackClassName={styles.sliderTrack}
                   thumbClassName={styles.sliderThumb}
                   renderThumb={(thumbProps, state) => (
                     <div {...thumbProps}>{state.valueNow}</div>
                   )}
                   defaultValue={slicingIndex}
-                  onChange={values => {
-                    if (values === undefined) {
+                  onChange={value => {
+                    if (!isNumber(value)) {
                       return;
                     }
-                    const newMapping: DimensionMapping = {
-                      x: mapping.x,
-                      slicingIndices: {
-                        ...mapping.slicingIndices,
-                        [Number(dimNumber)]: values as number,
-                      },
-                    };
-                    onChange(newMapping);
+                    const newMapperState = mapperState.slice();
+                    newMapperState[dim] = value;
+                    onChange(newMapperState);
                   }}
                   min={0}
-                  max={rawDims[Number(dimNumber)] - 1}
+                  max={rawDims[dim] - 1}
                   step={1}
                   orientation="vertical"
                   invert
                 />
-              </>
+              </Fragment>
+            ) : (
+              undefined
```

Comment:
Just for lisibility, it might be nicer to have a proper function body for the `map` arrow function: `=> { ... }` with an early return inside.